.github/workflows/ci.yml

```
@@ -56,22 +81,38 @@ jobs:
           arch: arm64
 
     runs-on: ${{ matrix.host }}
-    needs: build
+    needs: [build, build-32bit]
     steps:
     - name: install rust
+      if: ${{ matrix.target.arch == 'x64'}}
       uses: hecrj/setup-rust-action@v1
       with:
         rust-version: stable
+    
+    - name: install rust i686
+      if: ${{ matrix.target.arch == 'x86' }}
+      uses: hecrj/setup-rust-action@v1
+      with:
+        rust-version: stable
+        targets: "i686-unknown-linux-gnu"
 
     - name: install x
+      if: ${{ matrix.host-arch == 'x64'}}
       uses: actions/download-artifact@v3
       with:
         name: ${{ matrix.host }}-x
+
+    - name: install x i686
+      if: ${{ matrix.host-arch == 'x86'}}
+      uses: actions/download-artifact@v3
+      with:
+        name: i686-x
+
     - run: chmod +x ./x
       if: ${{ matrix.host == 'ubuntu-latest' || matrix.host == 'macos-latest' }}
 
     - name: install llvm ubuntu
-      if: ${{ matrix.host == 'ubuntu-latest' }}
+      if: ${{ matrix.host == 'ubuntu-latest'}}
```

Revert all unnecessary whitespace changes:

```suggestion
      if: ${{ matrix.host == 'ubuntu-latest' }}
```